lighthouse-core/lib/dependency-graph/simulator/connection-pool.js

```diff
@@ -114,21 +129,20 @@ module.exports = class ConnectionPool {
     const origin = String(record.parsedURL.securityOrigin);
     /** @type {TcpConnection[]} */
     const connections = this._connectionsByOrigin.get(origin) || [];
-    // Sort connections by decreasing congestion window, i.e. warmest to coldest
-    const availableConnections = connections
-      .filter(connection => !this._connectionsInUse.has(connection))
-      .sort((a, b) => b.congestionWindow - a.congestionWindow);
-
     const observedConnectionWasReused = !!this._connectionReusedByRequestId.get(record.requestId);
-
-    /** @type {TcpConnection|undefined} */
-    let connectionToUse = availableConnections[0];
-    if (!options.ignoreConnectionReused) {
-      connectionToUse = availableConnections.find(
-        connection => connection.isWarm() === observedConnectionWasReused
-      );
-    }
-
+    // Find the set of available connections based on the acquire options
+    const availableConnections = connections
+      .filter(connection => {
+        // Connections that are in use are never available
+        if (this._connectionsInUse.has(connection)) return false;
```

Comment:
maybe move this check to the end, since it's more work than the others. would be nice to defer.

Reply:
done